### DISCUSSION TITLE
Post-Processing status notification

### DISCUSSION THREAD

**Discussion by Juliocsreis** [codemagic-ci-cd/codemagic-docs#2754] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Build and deploy a Flutter app to App Store

### Expected results

Published app, or if fails for some reason send a slack notification

### Actual results

So, this only happened once so far, but publishing to App Store failed  as you can see in the image below. 

But CodeMagic sent a slack notification saying it was build successfully. 

Is it possible add a post-processing notification if the build fails ? 

Cause otherwise we thought the publishing was ok, and it actually was only ok on Google Play, which made our CEO really upset. 

<img width="705" alt="Captura de Tela 2024-09-24 às 12 42 05" src="https://github.com/user-attachments/assets/2b1d5b63-2ef0-4bbc-b883-af536a1268e7">

### Build id (optional)

66e1de83311ec415d549bcee

**Comment by dam-ease** [comment, marked as answer]
Hello @Juliocsreis,
The post processing flow is executed by App Store Connect which means that at the step, Codemagic has already completed its job and, now App Store should complete the rest. In case of any failures, Apple will send you an email explaining what went wrong and what is missing in the project.
Also, please note that it might take some time for Apple to send such emails. Can you please confirm that you got an email on this?

**Reply by Juliocsreis** [reply]
Hello @dam-ease . Thanks for taking the time to answer. 

No email was received from Apple related that. 

Do you think it's possible to edit codemagic.yaml to check the post-process status ?

**Reply by dam-ease** [reply]
Unfortunately, since the post-processing is managed by App Store Connect, Codemagic does not have direct visibility into the status once the build has been successfully handed off to Apple. This is why we can't directly check the post-process status within Codemagic at the moment.

From the error message, it seems like this might have been a temporary server-side issue on Apple's end, which could also explain why you didn't receive any email notification. If you try publishing again now, everything should go through smoothly.

**Reply by Juliocsreis** [reply]
I understand, but since CodeMagic got the logs from the process handed to Apple I believe it would be possible to check if the  response.statuts_code != 200, and then trigger a slack notification, no ?

**Reply by dam-ease** [reply]
Hi @Juliocsreis! Unfortunately, we do not have that feature available at the moment. However, you can always submit a feature request for it by clicking on this link: [here](https://github.com/orgs/codemagic-ci-cd/discussions/new?category=ideas). We are always open to new ideas and suggestions.
